app/models/layer.js

```diff
@@ -0,0 +1,137 @@
+import Model from 'ember-data/model';
+import { computed, set } from '@ember/object';
+import { alias } from '@ember/object/computed';
+import { copy } from 'ember-copy';
+
+import { assign } from '@ember/polyfills';
+import { next } from '@ember/runloop';
+import DS from 'ember-data';
+
+const { attr, belongsTo } = DS;
+
+/**
+  Model for individual layers. Belongs to a layer-group. May be called individually for state changes.
+
+  @public
+  @class LayerModel
+*/
+export default Model.extend({
+  init(...args) {
+    this._super(...args);
+
+    // enforce presence of blank object for mapboxGL validation
+    if (!this.get('style.layout')) this.set('style.layout', {});
+
+    // determine which is the first occurring layer
```

Comment:
`// Set current layer as first occuring layer for owning layer group if one does not already exist`

Reply:
Sorry, this is code ported from an addon. I needed to fix something in the addon. This was pure laziness, I'll go ahead and make the fix in the addon, publish a new version, and update.